staging/src/k8s.io/client-go/tools/cache/heap.go

```diff
@@ -251,9 +251,8 @@ func (h *Heap) Pop() (interface{}, error) {
 	obj := heap.Pop(h.data)
 	if obj != nil {
 		return obj, nil
-	} else {
-		return nil, fmt.Errorf("object was removed from heap data")
 	}
+	return nil, fmt.Errorf("object was removed from heap data")
```

Comment:
It's probably safe to compare `obj` against nil as it should be a pointer, not an interface. In that case we can swap the error flow into the check and leave the valid return in the main flow, to align the happy path with the least indentations.